src/main/java/com/dataguise/saas/service/SchedulerService.java

```diff
@@ -0,0 +1,434 @@
+                }
+            } catch (DgDrvException e) {
+                logger.error("Error updating the schedule.");
+                throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+            }
+
+        }
+        return status;
+    }
+
+    /**
+     * Get schedule event history by scheduleId
+     *
+     * @param scheduleId
+     */
+    public List<DgTaskSchedulerTaskHistBean> getScheduleHistory(Integer scheduleId)
+            throws BadRequestException, BadGatewayException {
+        List<DgTaskSchedulerTaskHistBean> taskSchedulerTaskHistBeanList = null;
+        if (scheduleId == null) {
+            throw new BadRequestException("Schedule "+ ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        try {
+            taskSchedulerTaskHistBeanList = restDriver.getTaskInstancesBySchedule(scheduleId);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return taskSchedulerTaskHistBeanList;
+    }
+
+    /**
+     * Get schedule summary by scheduleId
+     *
+     * @param scheduleId
+     */
+    public DgTaskScheduledExecutionsBean getScheduleSummary(Optional<Integer> scheduleId)
+            throws BadRequestException, BadGatewayException {
+        DgTaskScheduledExecutionsBean scheduleSummary = new DgTaskScheduledExecutionsBean();
+        try {
+            if (scheduleId.isPresent() && scheduleId.get() > 0) {
+                scheduleSummary = restDriver.getScheduleSummary(scheduleId.get());
+            } else {
+                scheduleSummary = restDriver.getScheduleSummary(0);
+            }
+        } catch (DgDrvException e) {
+            logger.error("Error retrieving the schedule summary"); // Not throwing an error in this case because GUI require default values
+            Long defaultSummary = new Long(0);
+            scheduleSummary.setFailedExecutions(defaultSummary);
+            scheduleSummary.setInProgressExecutions(defaultSummary);
+            scheduleSummary.setSuccessfulExecutions(defaultSummary);
+            scheduleSummary.setTotalDetectionScheduler(defaultSummary);
+            scheduleSummary.setTotalExecutions(defaultSummary);
+            scheduleSummary.setTotalMaskingScheduler(defaultSummary);
+        }
+        return scheduleSummary;
+    }
+
+
+    public SchedulerDTO formatScheduleDTO(SchedulerDTO schedulerDto) throws BadRequestException, InternalServerException, BadGatewayException {
+        String startDateTime = null;
+        String weekday = null;
+        List<String> weekdays = new ArrayList<>();
+        java.util.Date localZoneDate = null;
+
+        //TODO: Accept date in ISO 8601 format and convert it into controller readable format.
+        String isoInputDate = schedulerDto.getStartDateTime();
+        logger.info("Input ISO date from UI: " + isoInputDate);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/SchedulerService.java L380
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-879341007>*